pytorch_lightning/strategies/deepspeed.py

```diff
@@ -461,28 +461,33 @@ def zero_stage_3(self) -> bool:
         return self.config.get("zero_optimization") and self.config.get("zero_optimization").get("stage") == 3
 
     def _initialize_deepspeed_train(self, model):
+        optimizer, scheduler = None, None
         if "optimizer" in self.config:
             rank_zero_info(
                 "You have specified an optimizer and/or scheduler within the DeepSpeed config."
                 " It is recommended to define it in `LightningModule.configure_optimizers`."
             )
-            optimizer, lr_scheduler = None, _get_default_scheduler_config()
```

Comment:
you can use the dataclass here with defaults, then this avoids the None checks below and the code is as simple as before.

Reply:
The scheduler instance is a required argument of `LRSchedulerConfig` and I didn't want to change that and make it Optional. Otherwise, we open the typing can of worms of checking `assert is not None` everywhere.

Considering this, do you see a different way of writing this?

Reply:
I tried to make it optional so we a can do `LRSchedulerConfig(None)` here but then the problem is that elsewhere we would need to `assert scheduler_config.scheduler is not None` to make typing checks pass when accessing that attribute.

So no, I guess it can't go much simpler ..